src/deploy-cromwell-on-azure/Deployer.cs

```diff
@@ -1014,13 +1039,18 @@ private Task<IGenericResource> CreateAppInsightsResourceAsync()
         private Task<BatchAccount> CreateBatchAccountAsync()
         {
             return Execute(
-                $"Creating Batch Account: {configuration.BatchAccountName}...",
-                () => new BatchManagementClient(tokenCredentials) { SubscriptionId = configuration.SubscriptionId }
-                    .BatchAccount
-                    .CreateAsync(configuration.ResourceGroupName, configuration.BatchAccountName, new BatchAccountCreateParameters { Location = configuration.RegionName }, cts.Token)
+                $"Finding Existing Batch Account or Creating New Batch Account: {configuration.BatchAccountName}...",
+                async () => await TryGetExistingBatchAccountAsync() ?? await CreateNewBatchAccount()
                 );
         }
 
+        private Task<BatchAccount> CreateNewBatchAccount()
```

Comment:
How about this:
```
        private async Task<BatchAccount> GetOrCreateBatchAccountAsync()
        {
            var batchAccount = await GetExistingBatchAccountAsync(configuration.BatchAccountName);

            if (batchAccount != null)
            {
                return batchAccount;
            }

            return await Execute(
                $"Creating Batch Account: {configuration.BatchAccountName}...",
                () => new BatchManagementClient(tokenCredentials) { SubscriptionId = configuration.SubscriptionId }
                    .BatchAccount
                    .CreateAsync(configuration.ResourceGroupName, configuration.BatchAccountName, new BatchAccountCreateParameters { Location = configuration.RegionName }, cts.Token));
        }
```

Reply:
My current change is that I created the existing batch account object earlier in the process, and then will use it here.